pkg/agent/config/config.go

```diff
@@ -41,7 +42,7 @@ func Get(ctx context.Context, agent cmds.Agent, proxy proxy.Proxy) *config.Node
 	for {
 		agentConfig, err := get(ctx, &agent, proxy)
 		if err != nil {
-			logrus.Errorf("Failed to retrieve agent config: %v", err)
+			logrus.Errorf("Failed to configure agent: %v", err)
```

Comment:
Does this address an usability concern that you encountered while working on this feature? I understand that the `Get()` / `get()` are likely very poorly misnamed but this seems like a code smell: i.e. returning a "failed to do <do something>" message on an ostensible read operation.

Reply:
`get()` actually does more than just retrieve the config from the server - it retrieves a number of items from the server, merges in the CLI configuration, and validates the settings. The message here indicating that the **retrieval** failed was not accurate in many cases.